src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs

```diff
@@ -287,42 +287,42 @@ private async ValueTask<DifferenceViewerPreview> CreateAddedDocumentPreviewViewC
 
             // Create PreviewWorkspace around the buffer to be displayed in the diff preview
             // so that all IDE services (colorizer, squiggles etc.) light up in this buffer.
-            var rightWorkspace = new PreviewWorkspace(document.Project.Solution);
-            rightWorkspace.OpenDocument(document.Id, newBuffer.AsTextContainer());
+            using var rightWorkspace = PreviewWorkspace.Create(document.Project.Solution);
+            rightWorkspace.Target.OpenDocument(document.Id, newBuffer.AsTextContainer());
 
 #pragma warning disable CA2007 // Consider calling ConfigureAwait on the awaited task (containing method uses JTF)
             return await CreateAddedDocumentPreviewViewCoreAsync(newBuffer, rightWorkspace, document, zoomLevel, cancellationToken);
 #pragma warning restore CA2007 // Consider calling ConfigureAwait on the awaited task
```

Comment:
same points about returning something that captures a local which appears to be disposed when we get to end of scope.

Reply:
➡️ Covered by https://github.com/dotnet/roslyn/pull/68054/files#r1182949296